### ISSUE TITLE
Portal servlet changes from Swagger - what happened?

### ISSUE THREAD

**Issue by morungos** [cBioPortal/cbioportal#960]
Our Jetty-based deployment has been broken by the Swagger changes, especially from the additional URL map for the portal servlet added to the web.xml  in https://github.com/cBioPortal/cbioportal/commit/dfef8d12. We now get an error on start:

```
WARNING [org.eclipse.jetty.webapp.WebAppContext] Failed startup of context o.e.j.w.WebAppContext@50cbc42f{/,[file:/Users/stuartw/git/cbioportal-upstream/portal/target/portal/, jar:file:/Users/stuartw/git/cbioportal-upstream/portal/target/portal/WEB-INF/lib/springfox-swagger-ui-2.4.0.jar!/META-INF/resources],STARTING}
java.lang.IllegalStateException: Multiple servlets map to path: /: portal,default
```

This is because Jetty's default servlet, which exposes the JSPs, now has the same URL as the additional one exposed by the portal servlet in web.xml. I could also ask, should `rc` have two mappings for the `portal` servlet?

My question is (mainly for @sheridancbio, probably, as the Swagger guru) what is the right logic here?

**Comment by sheridancbio** [comment]
I'll try to learn what Jetty is and think of what might be altered in our servlet configuration to help debug the problem. I'll probably need help from a spring-mvc expert. (I'm still new to spring and swagger, but I'll try)

**Comment by morungos** [comment]
We can maybe schedule a time to debug if you like. I'm working on this now. I *think* I know how it works for e.g., tomcat, but I can't understand why. All my Swagger stuff is generally served as statics, so it shouldn't need a change to URL mapping.

**Comment by n1zea144** [comment]
Let me know if I can be of any assistance.  As I recall, when we were configuring SpringFox, we seemed to be locked into providing a mapping of '/' for it to work.  So we have '/' and '/api/*' both being mapped to our one Spring MVC DispatcherServlet, although our Controller only supports '/api/*'.

**Comment by sheridancbio** [comment]
Stuart suggested dropping the mapping to '/' in web.xml, and I have rebuilt with that change. The swagger JSON and swagger-ui.html both seem to be functional under /api/  (/api/api-docs and /api/swagger-ui.html). I'll test some more.

**Comment by n1zea144** [comment]
Well thats great.  Last time we checked, we couldn't get it to work without that mapping.  There must have been a change we made during the course of tuning the configuration that enabled that.  I can't think of what that may have been.  In any event, yeah, lets just make sure all the URLs still work.

**Comment by n1zea144** [comment]
@morungos @sheridancbio Is this resolved now?  Can we close it?

**Comment by morungos** [comment]
I'm good with that if the additional mapping -> / is merged into rc, as otherwise I'll need to mash up Jetty some more.

**Comment by sheridancbio** [comment]
I think it is resolved, but wanted to test a little more (make sure none of the other website functionality is affected). Calls to the Web API work, and the /api/api-docs and /api/swagger-ui.html both work, so odds are good that we can drop the mapping to '/'. When testing is done, I'll put this into a PR. If Stuart sees any problems in his local build (which does not have the '/' mapping) please let me know.

**Comment by sheridancbio** [comment]
Ok, testing is completed. I also forwarded the url "/api" to redirect to /api/swagger-ui.html

There was one issue I messed around with but could not figure out how to correct : the url "<appname>/swagger-ui.html" now exposes a "blank" swagger-ui page without our api references filled in. (and a link to "example.com"). I would have liked to have redirected requests to that to 
"<appname>/api/swagger-ui.html", but could not do it without breaking swagger. Any help on this would be welcome.
